File: clients/typescript/examples/collections-find-one-vector.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const collection = database.collection("**COLLECTION_NAME**");

// Find a document
(async function () {
  const result = await collection.findOne(
    {},
    { sort: { $vector: [0.08, -0.62, 0.39] } },
  );

  console.log(result);
})();

File: clients/python/examples/collections-find-one-vector.py
from astrapy import DataAPIClient

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
collection = database.get_collection("**COLLECTION_NAME**")

# Find a document
result = collection.find_one({}, sort={"$vector": [0.08, -0.62, 0.39]})

print(result)
